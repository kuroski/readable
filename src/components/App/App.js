import React from 'react';
import createReactClass from 'create-react-class';
import CategoriesBarContainer from '../../containers/CategoriesBarContainer/CategoriesBarContainer';
import './App.css';

const App = createReactClass({
  getInitialState: () => ({
    categories: []
  }),

  render: function() {
    return (
      <div className="App">
        <CategoriesBarContainer />
        {this.props.children}
      </div>
    );
  }
});

export default App;
