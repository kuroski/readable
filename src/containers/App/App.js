import React from 'react';
import createReactClass from 'create-react-class';
import CategoriesBar from '../../components/CategoriesBar/CategoriesBar';
import { Container } from 'semantic-ui-react';
import { Route } from 'react-router-dom';
import './App.css';

const App = createReactClass({
  getInitialState: () => ({
    categories: [
      {
        name: 'react',
        path: 'react'
      },
      {
        name: 'redux',
        path: 'redux'
      },
      {
        name: 'udacity',
        path: 'udacity'
      }
    ]
  }),

  render: function() {
    return (
      <div className="App">
        <CategoriesBar categories={this.state.categories} />
        <Container text style={{ marginTop: '7em' }}>
          <Route exact path="/" render={() => <h1>Ol√° mundo</h1>} />

          <Route
            path="/categories/:slug"
            render={({ match }) => <div>Categoria: {match.params.slug}</div>}
          />
        </Container>
      </div>
    );
  }
});

export default App;
