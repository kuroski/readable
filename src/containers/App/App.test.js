import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

it('renders without crashing', () => {
  const ShelvesTree = renderer
    .create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )
    .toJSON();
  expect(ShelvesTree).toMatchSnapshot();
});
