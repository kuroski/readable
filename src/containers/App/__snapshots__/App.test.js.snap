// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="ui inverted top fixed menu"
  >
    <div
      className="ui container"
    >
      <a
        aria-current="true"
        className="header item active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Readable
      </a>
      <a
        aria-current={false}
        className="item"
        href="/categories/react"
        onClick={[Function]}
        style={undefined}
      >
        react
      </a>
      <a
        aria-current={false}
        className="item"
        href="/categories/redux"
        onClick={[Function]}
        style={undefined}
      >
        redux
      </a>
      <a
        aria-current={false}
        className="item"
        href="/categories/udacity"
        onClick={[Function]}
        style={undefined}
      >
        udacity
      </a>
    </div>
  </div>
  <div
    className="ui text container"
    style={
      Object {
        "marginTop": "7em",
      }
    }
  >
    <h1>
      Ol√° mundo
    </h1>
  </div>
</div>
`;
